### DISCUSSION TITLE
Default LinearSNPCoverageDisplay for BAM track loaded by a UCSCTrackHubConnection

### DISCUSSION THREAD

**Discussion by GreenInnovator** [GMOD/jbrowse-components#2614] [Q&A]
To display a LinearSNPCoverageDisplay for a BAM track by default I can add to config.json:
```
      "displays": [
        {
          "type": "LinearSNPCoverageDisplay",
          "displayId": "example_snpcoverage"
        }
      ]
```
Is it possible to do the same, e.g. default LinearSNPCoverageDisplay for a BAM track, loaded by a UCSCTrackHubConnection? For example by a definition in trackDB.txt?

Regards, Jan

**Comment by garrettjstevens** [comment]
There isn't an option like that currently. Right now we only support some pretty basic options in the trackDb, but we could try to add some more. It looks like BAM/CRAM in the trackDb spec has a `doWiggle` setting that we might be able to interpret as changing the default display type to LinearSNPCoverageDisplay: https://genome.ucsc.edu/goldenPath/help/trackDb/trackDbHub.html#bam.

**Reply by GreenInnovator** [reply]
Another option might be `meta`

**Reply by garrettjstevens** [reply]
What is your use case? Does your data have to be in a track hub? We've considered creating a JBrowse 2 connection type where the tracks are specified in the native JBrowse configuration format, so you can customize them however you want.

**Reply by GreenInnovator** [reply]
Native JBrowse conf format sounds good. I have lots of projects with a lot of tracks. In Jbrowse 1 I used Jbrowse conf files that were included in the main file. 

There is no include option in JSON 2 format is there? I looked at add-json-track but that's one track at a time. I need a loop there.

**Reply by garrettjstevens** [reply]
There are no `include`s for the JBrowse 2 config. You could have a loop in the script where you're calling `jbrowse add-track` and add multiple tracks that way as well.
